Customer

JesperGreen opened the pull request:
PT Story: https://www.pivotaltracker.com/story/show/131079637

Changes proposed in this pull request:
- Created a user model (customers only so far).
- Installed devise (did not delete any devise views).
- Refactored erb views to haml.

What I have learned working on this feature: 
- How to refactor erb to haml via the Gemfile
- Installing devise, validate email/password
- How to combine cucumber steps in capybara

Screenshots: 
<img width="416" alt="screen shot 2016-09-29 at 5 54 46 pm" src="https://cloud.githubusercontent.com/assets/19630111/18961623/e29efcf0-866d-11e6-85d8-d8d10354905c.png">

<img width="856" alt="screen shot 2016-09-29 at 5 55 48 pm" src="https://cloud.githubusercontent.com/assets/19630111/18961654/036626ca-866e-11e6-82de-8f7b01fdc741.png">

Ready for review!

AmberWilkie committed:
Install Devise

AmberWilkie committed:
Add Devise boilerplate views

AmberWilkie committed:
Create migration for Devise user

AmberWilkie committed:
Add acceptance test for Customer registration

JesperGreen committed:
added fields to the sign-up form

JesperGreen committed:
added default values to user factory

JesperGreen committed:
added validation of role to user

AmberWilkie committed:
Overwrite Devise sanitation and routing for user registration

AmberWilkie committed:
Display current_user name on restaurant page

JesperGreen committed:
created user_steps.rb and refactored customer.feature

JesperGreen committed:
added route for sign up error 'name cant be blank'

JesperGreen committed:
added scenario for error when sign-up without email

JesperGreen committed:
addded scenario for sign-up error when address is blank

JesperGreen committed:
deleted unused method in registrations_controller

JesperGreen committed:
refactored erb views to haml

JesperGreen committed:
fixed merge conflicts

tochman commented on config/routes.rb:
```diff
@@ -1,7 +1,10 @@
   resources :menu, only: [:index, :create, :new]
   resources :dishes, only: [:new, :show, :create]
 
+  root to: 'restaurant#index'
```
If you add this route than I would remove line 3. It is redundant to keep both.

JesperGreen replied:
Fixed, I saved line 3 instead of line 8.
Line 8 wasn't actually working/doing anything.

tochman commented on features/customer.feature:
```diff
@@ -0,0 +1,39 @@
+
+Scenario: Gives you an error when you register without email
+  Given I register as a user with username "Jesper" and email ""
+  Then I should be on the "register error" page
```
I would call this `user registration` page

JesperGreen replied:
Yes, that makes sense.
Renamed it in all 3 scenarios.

tochman commented on features/customer.feature:
```diff
@@ -0,0 +1,39 @@
+
+Scenario: Gives you an error when you register without address
+  Given I register as a user with address ""
+  Then I should be on the "register error" page
```
Same here.

JesperGreen replied:
^above comment

tochman commented:
Good job. Small PR with a good scope. My only objections are some naming issues. Plz review that.

JesperGreen committed:
refactored config/routes.rb and renamed 'user registration page' for the PR

tochman commented on config/routes.rb:
```diff
@@ -5,6 +5,4 @@
   resources :menu, only: [:index, :create, :new]
   resources :dishes, only: [:new, :show, :create]
 
-  root to: 'restaurant#index'
```
I didn't mean for you to remove the `root_path` ;-)

AmberWilkie committed:
Address merge conflicts with upstream
fix failing cucumber tests following controller renames
fix large but ultimately easy overlap of routes

AmberWilkie committed:
Add Jesper's commits

AmberWilkie committed:
Remove double root route

tochman commented:
This became rather messy. Please stay on top regarding version control and also I prefer you guys to answer my comments instead of simply deleting code.

tochman commented:
We have conflicts on this now. Need to be fixed.

AmberWilkie committed:
Fix merge conflicts in application layout - flash msg display
